Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -1798,6 +1798,19 @@ void CGame::Packet_PlayerJoinData(CPlayerJoinDataPacket& Packet)
             return;
         }
 
+        // Check if another player is using the same serial
+        CPlayer* playerExisting = m_pPlayerManager->GetBySerial(strSerial);
+
+        if (playerExisting)
+        {
+            // Tell the console
+            CLogger::LogPrintf("CONNECT: %s failed to connect (Serial already in use) (%s)\n", szNick, strIPAndSerial.c_str());
+
+            // Tell the player the problem
+            DisconnectPlayer(this, *pPlayer, CPlayerDisconnectedPacket::SERIAL_DUPLICATE);
+            return;
+        }
```

Comment:
it would be better to have a const CPlayer* in an if statement:
```cpp
if (m_pPlayerManager->GetBySerial(strSerial))
{
    // Tell the console
    CLogger::LogPrintf("CONNECT: %s failed to connect (Serial already in use) (%s)\n", szNick, strIPAndSerial.c_str());

    // Tell the player the problem
    DisconnectPlayer(this, *pPlayer, CPlayerDisconnectedPacket::SERIAL_DUPLICATE);
    return;
}

// ...
const CPlayer* CPlayerManager::GetBySerial(const SString& serial) const noexcept
```

Reply:
I don't really understand this one, could you please explain?

Reply:
you dont need additional variable in CGame.cpp.
Just check it in the if statement

As for the GetBySerial function, you can return a const pointer to CPlayer. I mean, both would be fine

Reply:
I understand what you mean, it was just strange that you put the suggestion at the bottom of the code